src/Symfony/Component/Serializer/composer.json

```
@@ -17,7 +17,8 @@
     ],
     "require": {
         "php": ">=8.2",
-        "symfony/polyfill-ctype": "~1.8"
+        "symfony/polyfill-ctype": "~1.8",
+        "symfony/deprecation-contracts": "^3.4@dev"
```

That's a bit too strict. Also, the `@dev` is unnecessary.

```suggestion
        "symfony/deprecation-contracts": "^2.5|^3"
```

---

adjusted